pkg/authorization/plugin.go

```diff
@@ -72,12 +77,11 @@ func (a *authorizationPlugin) AuthZResponse(authReq *Request) (*Response, error)
 // initPlugin initializes the authorization plugin if needed
 func (a *authorizationPlugin) initPlugin() error {
 	// Lazy loading of plugins
-	if a.plugin == nil {
-		var err error
-		a.plugin, err = plugins.Get(a.name, AuthZApiImplements)
-		if err != nil {
-			return err
+	var err error
+	a.once.Do(func() {
+		if a.plugin == nil {
```

Comment:
it's possible in unit tests :P which I've struggled with the whole weekend and needs a big refactor to do discovery instead of just injecting the plugin directly in the struct so I deferred it to another PR I'll prepare